app/src/main/java/com/example/healthlog/handler/NewPatientHandler.java

```diff
@@ -327,17 +327,22 @@ String calculateAge() {
         Calendar dob = Calendar.getInstance();
         Calendar today = Calendar.getInstance();
 
-        dob.set(date.getYear(), date.getMonth(), date.getDay());
+        if(date == null){
+            return "Not available";
```

Comment:
Use string resources, also return "Date not selected properly" instead of "Not available"

Reply:
okay